src/test/java/com/synopsys/integration/alert/database/api/configuration/ClientCertificateAndKeyAccessorTestIT.java

```diff
@@ -0,0 +1,94 @@
+        ClientCertificateAndKeyModel getModel = accessor.getConfiguration()
+            .orElseThrow(() -> new AlertConfigurationException("Configuration does not exist"));
+
+        assertEquals(createdModel.getKeyPassword(), getModel.getKeyPassword());
+        assertEquals(createdModel.getKeyContent(), getModel.getKeyContent());
+        assertEquals(createdModel.getCertificateContent(), getModel.getCertificateContent());
+    }
+
+    @Test
+    void doesConfigurationExist() throws AlertConfigurationException {
+        Assertions.assertFalse(accessor.doesConfigurationExist());
+
+        accessor.createConfiguration(new ClientCertificateAndKeyModel("key_password", "key_content", "certificate_content"));
+        assertTrue(accessor.doesConfigurationExist());
+    }
+
+    @Test
+    void createConfigurationThrowsOnExistingConfig() throws AlertConfigurationException {
+        accessor.createConfiguration(new ClientCertificateAndKeyModel("key_password", "key_content", "certificate_content"));
+        assertTrue(accessor.doesConfigurationExist());
+
+        ClientCertificateAndKeyModel duplicateCreateModel =
+            new ClientCertificateAndKeyModel("new_key_password", "new_key_content", "new_certificate_content");
+        assertThrows(AlertConfigurationException.class, () -> accessor.createConfiguration(duplicateCreateModel));
+
+        ClientCertificateAndKeyModel getModel = accessor.getConfiguration()
+            .orElseThrow(() -> new AlertConfigurationException("Configuration does not exist"));
+
+        assertNotEquals(getModel.getKeyPassword(), duplicateCreateModel.getKeyPassword());
+        assertNotEquals(getModel.getKeyContent(), duplicateCreateModel.getKeyContent());
+        assertNotEquals(getModel.getCertificateContent(), duplicateCreateModel.getCertificateContent());
+    }
+
+    @Test
+    void updateConfiguration() throws AlertConfigurationException {
+        ClientCertificateAndKeyModel createdModel =
+            accessor.createConfiguration(new ClientCertificateAndKeyModel("key_password", "key_content", "certificate_content"));
+        ClientCertificateAndKeyModel updatedModel = accessor.updateConfiguration(
+            new ClientCertificateAndKeyModel(
+                "new_key_password",
+                "new_key_content",
+                "certificate_content"   // unchanged
+            )
+        );
+
+        Assertions.assertNotEquals(createdModel.getKeyPassword(), updatedModel.getKeyPassword());
+        Assertions.assertNotEquals(createdModel.getKeyContent(), updatedModel.getKeyContent());
+
+        Assertions.assertEquals(createdModel.getCertificateContent(), updatedModel.getCertificateContent());
+    }
+
+    @Test
+    void deleteConfiguration() throws AlertConfigurationException {
+        accessor.createConfiguration(new ClientCertificateAndKeyModel("key_password", "key_content", "certificate_content"));
+        accessor.deleteConfiguration();
```

Comment:
You were thorough on the exception case we should be consistent with deletion as well and make sure the configuration exists before trying to delete.

```suggestion
        Assertions.assertTrue(accessor.doesConfigurationExist());
        accessor.deleteConfiguration();
```